cmd/elementGen/main.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * MIT LICENSE
+ *
+ * Copyright © 2018, G.Ralph Kuntz, MD.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+// This generator should be run as "elementGen -o top_directory" or "go run cmd/elementGen/main.go".
```

Comment:
We want `elementGen` to be an [untyped `go generate`-or](https://github.com/myitcv/gogenerate/wiki/Go-Generate-Notes). That means we will almost certainly never invoke it by hand, rather it will be called by `go generate`. 

`go generate` is invoked as a result of directives in code. For example:

```go
//go:generate reactGen
```

is taken from [`react.go`](https://github.com/myitcv/react/blob/f32f0c5d7468766a687ac82cef960f5f64b72270/react.go#L14).

When `go generate` invokes a generator, it does so in the directory where the directive is found. So you can therefore rely on `os.Getwd()` to give you the information you need.

Other useful environment variables are set by `go generate` before it invokes the generator; see [`reactGen`](https://github.com/myitcv/react/blob/f32f0c5d7468766a687ac82cef960f5f64b72270/cmd/reactGen/react_gen.go#L52-L83) for an example, as well as the output from `go help generate`:

```
...

Go generate sets several variables when it runs the generator:

        $GOARCH
                The execution architecture (arm, amd64, etc.)
        $GOOS
                The execution operating system (linux, windows, etc.)
        $GOFILE
                The base name of the file.
        $GOLINE
                The line number of the directive in the source file.
        $GOPACKAGE
                The name of the package of the file containing the directive.
        $DOLLAR
                A dollar sign.

...
```